### PR TITLE
Use signatures and simplify common scripts

### PR DESCRIPTION
**Pull request by okurz**
* Fix language in check_needles.pl
* Simplify common scripts
* Use proper perl "signatures" in common script files

### CODE REVIEW

File: os-autoinst-openvswitch [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
 use autodie ':all';
 use Net::DBus::Exporter 'org.opensuse.os_autoinst.switch';
 require IPC::System::Simple;
 use IPC::Open3;
 use Symbol 'gensym';
 use Time::Seconds;
 
-sub new {
-    my $class = shift;
+sub new ($class) {
```
Comment (perlpunk) [inline os-autoinst-openvswitch:12]: You missed the `my $service = shift;` in the next line.

File: os-autoinst-openvswitch [hunk 2]
```diff
@@ -96,8 +94,7 @@ sub init_switch {
 }
 
 # Check if tap and vlan are in the correct format.
-sub _ovs_check {
-    my $tap = shift;
+sub _ovs_check ($tap) {
```
Comment (perlpunk) [inline os-autoinst-openvswitch:97]: similar here

File: os-autoinst-openvswitch [hunk 3]
```diff
@@ -151,8 +146,7 @@ sub check_min_ovs_version {
 }
 
 dbus_method("set_vlan", ["string", "uint32"], ["int32", "string"]);
-sub set_vlan {
-    my $self = shift;
+sub set_vlan ($self) {
```
Comment (perlpunk) [inline os-autoinst-openvswitch:149]: ...and here

File: os-autoinst-openvswitch [hunk 4]
```diff
@@ -185,8 +179,7 @@ sub set_vlan {
 }
 
 dbus_method("unset_vlan", ["string", "uint32"], ["int32", "string"]);
-sub unset_vlan {
-    my $self = shift;
+sub unset_vlan ($self) {
```
Comment (perlpunk) [inline os-autoinst-openvswitch:182]: and here

**Review by perlpunk — APPROVED** [review]

**Review by baierjan — APPROVED** [review]

**Pull request merged**